manifests/00-custom-resource-definition.yaml

```diff
@@ -442,22 +88,23 @@ spec:
                 hostNetwork:
                   description: hostNetwork holds parameters for the HostNetwork endpoint
                     publishing strategy. Present only if type is HostNetwork.
+                  nullable: true
                   type: object
                 loadBalancer:
                   description: loadBalancer holds parameters for the load balancer.
                     Present only if type is LoadBalancerService.
+                  nullable: true
                   properties:
                     scope:
                       description: scope indicates the scope at which the load balancer
                         is exposed. Possible values are "External" and "Internal".  The
                         default is "External".
                       type: string
-                  required:
-                  - scope
```

Comment:
this looks suspicious.

Reply:
4.1 didn't have `scope`, so it cannot be required.

Reply:
It's required in the 4.2 branch... https://github.com/openshift/cluster-ingress-operator/blob/release-4.2/manifests/00-custom-resource-definition.yaml#L455

Reply:
@Miciah https://github.com/openshift/cluster-ingress-operator/blob/release-4.2/manifests/00-custom-resource-definition.yaml#L455 does this mean we need to change 4.2 before release? Is there a problem that has gone unnoticed?

Reply:
Seems like this would be a problem.  I can understand how we would have missed this for `spec.endpointPublishingStrategy.scope`: `spec.endpointPublishingStrategy` is optional, and the operator-created default ingresscontroller does not specify it.  However, I wonder why we have not seen problems with `status.endpointPublishingStrategy.scope` during upgrade tests.  I'll launch a 4.1 cluster, set `spec.endpointPublishingStrategy` with null `spec.endpointPublishingStrategy.status`, upgrade the cluster to 4.2, and report back what happens.

Reply:
@Miciah based on your upgrade testing, seems like this is going to be okay?

Reply:
Right, the changing `required` value did not seem to have any effect: I performed the steps I laid out in my previous comment and saw no errors in the operator logs or command output when the upgraded ingress operator reconciled the old IngressController resource that had `spec.endpointPublishingStrategy` but not `spec.endpointPublishingStrategy.scope` set and when I tried to update the resource using `oc annotate`.

Reply:
Facepalm: `endpointPublishingStrategy.loadBalancer` is optional, so `endpointPublishingStrategy.loadBalancer.scope` can be required.  I erroneously wrote `endpointPublishingStrategy.scope` instead of `endpointPublishingStrategy.loadBalancer.scope` earlier.  I think 4.2 is fine to make `scope` required, and we should keep it required in 4.3.

Reply:
https://github.com/openshift/api/pull/446 will decide the fate of this field

Reply:
Fixed.